typings/externs.d.ts

```diff
@@ -102,8 +102,17 @@ declare global {
       friendlyMessage?: string;
     }
 
+    /**
+     * A record of DevTools Debugging Protocol events.
+     */
+    export type DevtoolsLog = Array<Protocol.RawEventMessage>;
+
     export interface Trace {
       traceEvents: TraceEvent[];
+      metadata?: {
+        'cpu-family'?: number;
+      };
+      [futureProps: string]: any;
```

Comment:
do we need this added? afaik there aren't any other toplevel items in a trace than these two.

Reply:
> do we need this added?

we support [outputting arbitrary keys in the trace object](https://github.com/GoogleChrome/lighthouse/blob/aec55bd59b39aeca7a84f51b760fcd4626897b25/lighthouse-core/lib/asset-saver.js#L211-L218) in `saveTrace()` and `logAssets()`, which seems worth keeping?

But then tsc needs to know there are other properties to loop over in the first place, so it needs to be either ts-ignored in place or a generic "every other property name" entry needs to be added here.

So...what do you think we should do

Reply:
ah ok. well that's fine. can you just leave a comment that this is for our flexible traceWriter method?